File: android/app/src/main/kotlin/com/navitech/cabo/support/FeedbackSupport.kt
package com.navitech.cabo.support

import android.content.ActivityNotFoundException
import android.content.Context
import android.content.Intent
import android.net.Uri
import android.os.Build
import com.navitech.cabo.BuildConfig

/** Same inbox as iOS `FeedbackSupport.recipientEmail`. */
object FeedbackSupport {
    const val RECIPIENT_EMAIL = "f.alexandru577@gmail.com"

    fun diagnosticFooter(): String {
        val ver = BuildConfig.VERSION_NAME
        val code = BuildConfig.VERSION_CODE
        val sdk = Build.VERSION.SDK_INT
        val model = Build.MODEL ?: "?"
        return "\n\n—\nVersion $ver ($code)\nAndroid API $sdk, $model"
    }

    /**
     * Opens a mail handler with subject/body prefilled. Returns false if no app handles mailto.
     */
    fun openFeedbackEmail(context: Context): Boolean {
        val subject = "Cabo feedback"
        val body = diagnosticFooter()
        val uri =
            Uri.parse(
                "mailto:$RECIPIENT_EMAIL?subject=${Uri.encode(subject)}&body=${Uri.encode(body)}"
            )
        val intent =
            Intent(Intent.ACTION_SENDTO).apply {
                data = uri
            }
        return try {
            context.startActivity(Intent.createChooser(intent, "Send feedback"))
            true
        } catch (_: ActivityNotFoundException) {
            false
        }
    }
}

File: ios/CaboGame/Support/FeedbackSupport.swift
import Foundation
import UIKit

/// Address shown in Mail / mailto.
enum FeedbackSupport {
    static let recipientEmail = "f.alexandru577@gmail.com"

    static var diagnosticFooter: String {
        let ver = Bundle.main.object(forInfoDictionaryKey: "CFBundleShortVersionString") as? String ?? "?"
        let build = Bundle.main.object(forInfoDictionaryKey: "CFBundleVersion") as? String ?? "?"
        let os = UIDevice.current.systemVersion
        let model = UIDevice.current.model
        return "\n\n—\nVersion \(ver) (\(build))\niOS \(os), \(model)"
    }

    static func mailtoURL(subject: String = "Cabo feedback") -> URL? {
        var c = URLComponents()
        c.scheme = "mailto"
        c.path = recipientEmail
        c.queryItems = [
            URLQueryItem(name: "subject", value: subject),
            URLQueryItem(name: "body", value: diagnosticFooter),
        ]
        return c.url
    }
}
